scripts/prepare

```diff
@@ -6,35 +6,37 @@ set -e
 
 for f in packages/*; do
 	if [[ -f ${f}/package.yaml ]]; then
-		if [[ -z $CHART || $CHART == $(basename -- ${f}) ]]; then
-			url=$(cat ${f}/package.yaml | yq r - url)
-			subdirectory=$(cat ${f}/package.yaml | yq r - subdirectory)
-			type=$(cat ${f}/package.yaml | yq r - type)
-			fields=$(echo ${subdirectory} | awk -F'/' '{print NF}')
-			commit=$(cat ${f}/package.yaml | yq r - commit)
-			if [[ $fields -eq '0' ]]; then
-				fields='1'
-			fi
-			rm -rf ${f}/charts
-			mkdir -p ${f}/charts
-			if [[ $type == 'git' ]]; then
-				mkdir -p /tmp/tmp-charts
-				git clone --depth=1 --no-tags $url /tmp/tmp-charts
-				pwd=$(pwd)
-				cd /tmp/tmp-charts
-				git fetch origin $commit
-				git checkout $commit
-				cd $pwd
+		if [[ (-z $CHART || $CHART == $(basename -- ${f})) ]]; then
+			if [[ $(cat ${f}/package.yaml | yq r - url) ]]; then
+				url=$(cat ${f}/package.yaml | yq r - url)
+				subdirectory=$(cat ${f}/package.yaml | yq r - subdirectory)
+				type=$(cat ${f}/package.yaml | yq r - type)
+				fields=$(echo ${subdirectory} | awk -F'/' '{print NF}')
+				commit=$(cat ${f}/package.yaml | yq r - commit)
+				if [[ $fields -eq '0' ]]; then
+					fields='1'
+				fi
 				rm -rf ${f}/charts
-				cp -r /tmp/tmp-charts/${subdirectory} ${f}/charts
-				rm -rf /tmp/tmp-charts
-			else
-				curl -sLf ${url} | tar xvzf - -C ${f}/charts --strip ${fields} ${subdirectory} > /dev/null 2>&1
+				mkdir -p ${f}/charts
+				if [[ $type == 'git' ]]; then
+					mkdir -p /tmp/tmp-charts
+					git clone --depth=1 --no-tags $url /tmp/tmp-charts
+					pwd=$(pwd)
+					cd /tmp/tmp-charts
+					git fetch origin $commit
+					git checkout $commit
+					cd $pwd
+					rm -rf ${f}/charts
+					cp -r /tmp/tmp-charts/${subdirectory} ${f}/charts
+					rm -rf /tmp/tmp-charts
+				else
+					curl -sLf ${url} | tar xvzf - -C ${f}/charts --strip ${fields} ${subdirectory} >/dev/null 2>&1
+				fi
+				for file in $(find ./${f} -type f -name "*.patch"); do
+					basename=$(basename -- ${file})
+					patch -E -p3 -d ${f}/charts <${f}/$basename
+				done
 			fi
-			for file in $(find ./${f} -type f -name "*.patch"); do
-				basename=$(basename -- ${file})
-				patch -E -p3 -d ${f}/charts < ${f}/$basename
-			done
```

Comment:
Git's diff here is hard to read, are the only changes the wrapping in the conditional on line 10, and moving line 34-37 up ?

Reply:
yes